### ISSUE TITLE
Is HIP support planned for cpu only?

### ISSUE THREAD

**Issue by ghost** [WolframRhodium/VapourSynth-BM3DCUDA#21]
Or will hip capable gpu cards will be able to use it?

**Comment by WolframRhodium** [comment]
HIP capable consumer GPU should be able to use it.

**Comment by LuisB79** [comment]
> HIP capable consumer GPU should be able to use it.

Based, never thought you'd actually do it, is it ready for testing?, im eager to find bugs

**Comment by WolframRhodium** [comment]
Yes it is ready for testing. On Linux there is a possibility of SIGSEGV error with default `fast=True`, not sure whether that is the case on Windows.

**Comment by quietvoid** [comment]
> On Linux there is a possibility of SIGSEGV error with default fast=True

I just ran into this :slightly_frowning_face:

**Comment by WolframRhodium** [comment]
> I just ran into this 🙁 Fortunately for my use cases there doesn't seem to be a real performance issue with using `fast=False`.

They need to fix it, probably relate to multi-threaded graph execution.
